app/leagues/page.tsx

```diff
@@ -3,52 +3,47 @@
 
 import React, { JSX } from 'react';
 import { LeagueCard } from '@/components/LeagueCard/LeagueCard';
+import { IUser } from '@/api/apiFunctions.interface';
+import { getCurrentUser } from '@/api/apiFunctions';
+import { getUserLeagues } from '@/utils/utils';
+
+interface Props {
+  searchParams?: { id: IUser['id'] };
+}
 
 /**
  * Renders the leagues component.
- * @returns {JSX.Element} The rendered leagues component.
+ * @param {Props} props The props for the leagues component.
+ * @returns {JSX.Element | undefined} The rendered leagues component.
  */
-const Leagues = (): JSX.Element => {
+const Leagues = async ({
+  searchParams,
+}: Props): Promise<JSX.Element | undefined> => {
+  const userId = searchParams?.id || '';
+
+  if (!userId || userId === '') {
+    return;
+  }
+
+  const userData: IUser = await getCurrentUser(userId);
+  const leagueData = await getUserLeagues(userData.league);
+
   return (
     <div className="Leagues mx-auto max-w-3xl pt-10">
       <h1 className="pb-10 text-center text-3xl font-bold tracking-tight">
         Your leagues
       </h1>
       <section className="grid gap-6 md:grid-cols-2">
-        <LeagueCard
-          href="/leagues"
-          leagueCardLogo="https://ryanfurrer.com/_astro/logo-dark-theme.CS8e9u7V_JfowQ.svg" // should eventually be something like league.logo
-          survivors={69}
-          title="69ers"
-          totalPlayers={138}
-        />
-        <LeagueCard
-          isEliminated={true}
-          href="/leagues"
-          survivors={69}
-          title="69ers"
-          totalPlayers={138}
-        />
-
-        <LeagueCard
-          href="/leagues"
-          survivors={11}
-          title="It's Always Sunny in Minneapolis"
-          totalPlayers={12}
-        />
-        <LeagueCard
-          href="/leagues"
-          survivors={20}
-          title="Windy City Smackdown"
-          totalPlayers={24}
-        />
-        <LeagueCard
-          href="/leagues"
-          isEliminated={true}
-          survivors={11}
-          title="Diamond Dogs"
-          totalPlayers={12}
-        />
+        {leagueData.map((league) => (
```

Comment:
We should check that leagueData length is greater than 0. If not, show a message that says `You are not particiapting in any leagues`

Reply:
✅ Done.

```js
        {leagueData.length > 0 ? (
          leagueData.map((league) => (
            <LeagueCard
              key={league.leagueId}
              href={`/picks?leagueId=${league.leagueId}`}
              leagueCardLogo="https://ryanfurrer.com/_astro/logo-dark-theme.CS8e9u7V_JfowQ.svg" // should eventually be something like league.logo
              survivors={league.survivors.length}
              title={league.leagueName}
              totalPlayers={league.participants.length}
            />
          ))
        ) : (
          <div className="text-center">
            <p className="text-lg font-bold">
              You are not participating in any leagues.
            </p>
          </div>
        )}
```